File: MultiLanguageSandbox/src/test/deps/typescript/23.ts

/**
 * Count the number of acute-angled triangles that can be formed by selecting any 3 edges out of a given set of edges.
 * 
 * @param edges An array of integers representing the lengths of the edges.
 * @returns The count of distinct acute-angled triangles that can be formed.
 * 
 * Examples:
 * ```
 * countAcuteTriangles([1, 1, 1, 1])
 * // returns 4
 * ```
 */

function countAcuteTriangles(edges: number[]): number {
    let count = 0;
    const n = edges.length;
    
    // Iterate through all possible triplets
    for (let i = 0; i < n; i++) {
        for (let j = i + 1; j < n; j++) {
            for (let k = j + 1; k < n; k++) {
                const a = edges[i];
                const b = edges[j];
                const c = edges[k];
                
                // Check if they can form a triangle
                if (a + b > c && a + c > b && b + c > a) {
                    // Check if the triangle is acute
                    const aSquared = a * a;
                    const bSquared = b * b;
                    const cSquared = c * c;
                    
                    if (aSquared + bSquared > cSquared && 
                        aSquared + cSquared > bSquared && 
                        bSquared + cSquared > aSquared) {
                        count++;
                    }
                }
            }
        }
    }
    
    return count;
}
(() => {
    // Test function using console.assert for validation
    console.assert(countAcuteTriangles([1, 1, 1, 1]) === 4, "Test 1 Failed");
    console.assert(countAcuteTriangles([1, 2, 3]) === 0, "Test 2 Failed");
    // Additional tests to ensure correctness
    console.assert(countAcuteTriangles([3, 4, 5, 7, 10]) === 0, "Test 3 Failed");
    console.assert(countAcuteTriangles([6, 8, 10, 5, 5, 5]) === 4, "Test 4 Failed");
    // console.log("All tests passed!");
})();

File: MultiLanguageSandbox/src/test/deps/pascal/23.pas
{$ASSERTIONS ON}

uses
  Math, SysUtils;
(*
  Determines the number of acute-angled triangles that can be formed from a given set of edge lengths.
  
  Parameters:
  - edges: An open array of integers representing edge lengths.

  Returns:
  The count of distinct, valid acute-angled triangles that can be formed.
  
  Examples:
  writeln(count_acute_triangles([1, 1, 1, 1])); (* Output: 4 *)
*)

function count_acute_triangles(var edges: array of integer): integer;
var
  i, j, k, n, count: integer;
  a, b, c: integer;
begin
  n := Length(edges);
  count := 0;
  
  // Sort the array to facilitate checking conditions
  for i := 0 to n - 2 do
    for j := i + 1 to n - 1 do
      if edges[i] > edges[j] then
      begin
        a := edges[i];
        edges[i] := edges[j];
        edges[j] := a;
      end;
  
  // Iterate through all possible triplets
  for i := 0 to n - 3 do
    for j := i + 1 to n - 2 do
      for k := j + 1 to n - 1 do
      begin
        a := edges[i];
        b := edges[j];
        c := edges[k];
        
        // Check triangle inequality: a + b > c
        if a + b > c then
        begin
          // Check for acute triangle: a^2 + b^2 > c^2
          if (a * a + b * b) > (c * c) then
            Inc(count);
        end;
      end;
  
  Result := count;
end;
procedure testCountAcuteTriangles;
var
  test_array: array of integer;
begin
  SetLength(test_array, 4);test_array[0] := 1;test_array[1] := 1;test_array[2] := 1;test_array[3] := 1;
  Assert(count_acute_triangles(test_array) = 4);
  SetLength(test_array, 3);test_array[0] := 1;test_array[1] := 2;test_array[2] := 3; 
  Assert(count_acute_triangles(test_array) = 0);
  SetLength(test_array, 5);test_array[0] := 3;test_array[1] := 4;test_array[2] := 5;test_array[3] := 7;test_array[4] := 10;
  Assert(count_acute_triangles(test_array) = 0);
  SetLength(test_array, 6);test_array[0] := 6;test_array[1] := 8;test_array[2] := 10;test_array[3] := 5;test_array[4] := 5;test_array[5] := 5;
  Assert(count_acute_triangles(test_array) = 4);
end;

begin
  testCountAcuteTriangles;
end.